src/components/welcome/welcome.test.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { render, waitFor } from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+import Welcome from './welcome'
+
+global.fetch = require('jest-fetch-mock')
+const apiStringResponse = 'Welcome to the Gilded Rose LLC store!'
+
+test('Welcome message is printed in the <Welcome /> component upon componentDidMount', async () => {
+  fetch.mockResponseOnce(JSON.stringify())
+  const { getByTestId } = render(<Welcome welcomeString={ apiStringResponse }/>)
+  await waitFor(() => getByTestId('welcome'))
+
+  expect(getByTestId('welcome').textContent).toBe('Welcome to the Gilded Rose LLC store!')
+})
```

Comment:
This is related to one of my other comments from earlier. Can you add a test for the error? Like 'An error message is printed to the screen if the page fails to retrieve data from the API'? Also, can you add a test for loading? Like 'A loading message is printed to the screen if the page is loading'